src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3116,15 +3116,18 @@ void WriteErrorHelper(Exception exception)
                             _removedFiles++;
                             _removedBytes += fileBytesSize;
                             double speed = _removedBytes / 1024 / 1024 / _removeStopwatch.Elapsed.TotalSeconds;
-                            var progress = new ProgressRecord(
-                                REMOVE_FILE_ACTIVITY_ID,
-                                StringUtil.Format(FileSystemProviderStrings.RemovingLocalFileActivity, _removedFiles, _totalFiles),
-                                StringUtil.Format(FileSystemProviderStrings.RemovingLocalBytesStatus, Utils.DisplayHumanReadableFileSize(_removedBytes), Utils.DisplayHumanReadableFileSize(_totalBytes), speed)
-                            );
-                            var percentComplete = _totalBytes != 0 ? (int)Math.Min(_removedBytes * 100 / _totalBytes, 100) : 100;
-                            progress.PercentComplete = percentComplete;
-                            progress.RecordType = ProgressRecordType.Processing;
-                            WriteProgress(progress);
+                            if (_removeStopwatch.Elapsed.TotalSeconds > ProgressBarDurationThreshold)
```

Comment:
`double speed = ...` should be moved under follow `if`.

Reply:
resolved